### PR TITLE
Add LostPassword Flood Control

### LINKED ISSUE #3887: Use FloodControl in LostPasswordForm

### PR DESCRIPTION
**Pull request by joshuaruesweg**
Closes #3887

### CODE REVIEW

File: wcfsetup/install/lang/de.xml [hunk 1]
```diff
@@ -4600,6 +4600,7 @@ Dateianhänge:
...
 		<item name="wcf.user.lostPassword.email.error.notFound"><![CDATA[Es wurde kein Benutzer mit der E-Mail-Adresse: „{$email}“ gefunden.]]></item>
 		<item name="wcf.user.lostPassword.error.tooManyRequests"><![CDATA[Das Kennwort für dieses Benutzerkonto wurde in den letzten 24 Stunden bereits einmal angefordert. Aus Sicherheitsgründen kann das Kennwort eines Benutzers nur einmal pro Tag angefordert werden. {if LANGUAGE_USE_INFORMAL_VARIANT}Du kannst{else}Sie können{/if} das Kennwort für dieses Benutzerkonto in {#$hours} Stunde{if $hours != 1}n{/if} erneut anfordern.]]></item>
+		<item name="wcf.user.lostPassword.error.flood"><![CDATA[{if LANGUAGE_USE_INFORMAL_VARIANT}Du hast{else}Sie haben{/if} zu viele Kennwörter angefordert. Die Funktion wurde daher aus Sicherheitsgründen temporär deaktiviert. Bitte {if LANGUAGE_USE_INFORMAL_VARIANT}versuche es{else}versuchen Sie es{/if} später erneut.]]></item>
```
Comment (joshuaruesweg) [inline wcfsetup/install/lang/de.xml:4603]: Is this fine? Better ideas for wording?

File: wcfsetup/install/files/lib/form/LostPasswordForm.class.php [hunk 2]
```diff
@@ -73,6 +75,10 @@ public function validate()
     {
         parent::validate();
 
+        if (FloodControl::getInstance()->countContent('com.woltlab.wcf.lostPasswordForm', new \DateInterval('PT24H'))['count'] >= self::ALLOWED_RESETS_PER_24H) {
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/form/LostPasswordForm.class.php:78]: If I need to scroll horizontally then the line is too long. Keep it at 120 characters, please.

File: wcfsetup/install/files/lib/form/LostPasswordForm.class.php [hunk 3]
```diff
@@ -27,6 +28,7 @@
 class LostPasswordForm extends AbstractCaptchaForm
 {
     const AVAILABLE_DURING_OFFLINE_MODE = true;
+    const ALLOWED_RESETS_PER_24H = 3;
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/form/LostPasswordForm.class.php:31]: Add constant visibility (`private`).

Comment (TimWolla) [inline wcfsetup/install/files/lib/form/LostPasswordForm.class.php:31]: Maybe increase this a little:
```suggestion
    private const ALLOWED_RESETS_PER_24H = 5;
```

**Review by TimWolla — CHANGES_REQUESTED** [review]

File: wcfsetup/install/lang/de.xml (hunk 1 above)
Comment (Gravatronics) [inline wcfsetup/install/lang/de.xml:4603]:
```suggestion
		<item name="wcf.user.lostPassword.error.flood"><![CDATA[{if LANGUAGE_USE_INFORMAL_VARIANT}Du hast{else}Sie haben{/if} zu viele neue Kennwörter angefordert. Die Funktion wurde daher aus Sicherheitsgründen temporär deaktiviert. Bitte {if LANGUAGE_USE_INFORMAL_VARIANT}versuche{else}versuchen Sie{/if} es später erneut.]]></item>
```

?

**Review by TimWolla — APPROVED** [review]

**Pull request merged**